boa_engine/src/syntax/parser/expression/primary/object_initializer/mod.rs

```diff
@@ -173,17 +174,31 @@ where
                 cursor.peek_expect_no_lineterminator(0, "Async object methods", interner)?;
 
                 let token = cursor.peek(0, interner)?.ok_or(ParseError::AbruptEnd)?;
+                let position = token.span().start();
+
                 if let TokenKind::Punctuator(Punctuator::Mul) = token.kind() {
                     let (property_name, method) =
                         AsyncGeneratorMethod::new(self.allow_yield, self.allow_await)
                             .parse(cursor, interner)?;
+
+                    // It is a Syntax Error if HasDirectSuper of MethodDefinition is true.
```

Comment:
```suggestion
                    // It is a Syntax Error if `HasDirectSuper` of `MethodDefinition` is `true`.
```

Reply:
Also, should this be named `has_direct_super()` instead of `function_contains_super_call()`?

Reply:
That is very true. It's basically that function in the spec.

I have not added the backticks for these comments, as they are just copied from the spec. I think it makes less sense on those.